### PR TITLE
[AIRFLOW-6346] Enhance dag default_view and orientation

### PR DESCRIPTION
**Pull request by zhongjiajie**
* Only DEFAULT_VIEW_PRESETS are acceptable
* Only ORIENTATION_PRESETS are acceptable
* Use webserver.dag_default_view in airflow.cfg as default value
* Remove DAG and DagModel function get_default_view
* Rename _default_value to default_value

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6346

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

* Only DEFAULT_VIEW_PRESETS are acceptable
* Only ORIENTATION_PRESETS are acceptable
* Use webserver.dag_default_view in airflow.cfg as default value
* Remove DAG and DagModel function get_default_view
* Rename _default_value to default_value

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by zhongjiajie** [comment]
Invalid `default_view` will cause Airflow webserver error.

File: airflow/models/dag.py [hunk 1]
```diff
@@ -296,8 +298,16 @@ def __init__(
         self.max_active_runs = max_active_runs
         self.dagrun_timeout = dagrun_timeout
         self.sla_miss_callback = sla_miss_callback
-        self._default_view = default_view
```
Comment (zhongjiajie) [inline airflow/models/dag.py:307]: I change name from `_default_view` -> `default_view`
I'm not really sure why we use `self._default_view` as property name before? Prevent cover the Flask `default_view` variable ?

**Comment by zhongjiajie** [comment]
CI green, PTAL @potiuk @mik-laj @kaxil

**Comment by potiuk** [comment]
I am not as good/knowledgable in the UI/Flask part. MAybe someone else can review this one?

File: airflow/models/dag.py (hunk 1 above)
Comment (kaxil) [reply airflow/models/dag.py:307]: I have no idea tbh, @vardancse @milton0825 - Do you guys have any idea about it?

Comment (milton0825) [reply airflow/models/dag.py:307]: no idea either...

Comment (zhongjiajie) [reply airflow/models/dag.py:307]: So I think is ok to change to `default_view `

**Comment by zhongjiajie** [comment]
According to https://github.com/apache/airflow/pull/6900#discussion_r363106891 no one know the reason,  @potiuk @kaxil I think is ok to change property name from `_default_view` to `default_view`, and I test it locally and it work, So pls task a look a this PR

**Comment by kaxil** [comment]
@ashb WDYT?

File: airflow/models/dag.py [hunk 2]
```diff
@@ -300,8 +302,16 @@ def __init__(
...
-        self.orientation = orientation
+        if default_view in DEFAULT_VIEW_PRESETS:
+            self.default_view = default_view
+        else:
+            raise AirflowException(f'Invalid values of dag.default_view: only support '
+                                   f'{DEFAULT_VIEW_PRESETS}, but get {default_view}')
```
Comment (ashb) [inline airflow/models/dag.py:313]: Do f-strings work in py3.6?

File: airflow/serialization/schema.json [hunk 3]
```diff
@@ -93,7 +93,7 @@
         "end_date": { "$ref": "#/definitions/datetime" },
         "dagrun_timeout": { "$ref": "#/definitions/timedelta" },
         "doc_md": { "type" : "string"},
-        "_default_view": { "type" : "string"},
+        "default_view": { "type" : "string"},
```
Comment (ashb) [inline airflow/serialization/schema.json]: This makes this version 2 of the schema sadly, so we now need to think about how we handle versions

**Review by ashb — CHANGES_REQUESTED** [review]

File: airflow/serialization/serialized_objects.py [hunk 4]
```diff
@@ -467,7 +467,7 @@ def __get_constructor_defaults():  # pylint: disable=no-method-argument
         param_to_attr = {
             'concurrency': '_concurrency',
             'description': '_description',
-            'default_view': '_default_view',
+            'default_view': 'default_view',
             'access_control': '_access_control',
```
Comment (ashb) [inline airflow/serialization/serialized_objects.py]: I don't think this is needed anymore and the whole line can be removed

File: airflow/models/dag.py (hunk 2 above)
Comment (zhongjiajie) [reply airflow/models/dag.py:313]: Yeah, I think it work in py3.6 https://www.python.org/dev/peps/pep-0498/

File: airflow/serialization/schema.json (hunk 3 above)
Comment (zhongjiajie) [reply airflow/serialization/schema.json]: Do you mean we should update user's exists schema in databases?

Comment (kaxil) [reply airflow/serialization/schema.json]: Yes, it would break the existing Dags in D. To get past it wr can revert the renaming of _default_view

Comment (zhongjiajie) [reply airflow/serialization/schema.json]: I remark here, so we have two ways:
* change schema name to `default_view` and make compatible to existing dags 
* keep in the old name `_default_view`

Comment (kaxil) [reply airflow/serialization/schema.json]: **keep in the old name _default_view** for now

Comment (zhongjiajie) [reply airflow/serialization/schema.json]: > **keep in the old name _default_view** for now

Ok, I get it.

**Comment by zhongjiajie** [comment]
@ashb @kaxil CI green, I think is ready to review

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Thanks @zhongjiajie

**Comment by zhongjiajie** [comment]
Thanks for merging @kaxil

**Pull request merged**